core/lib/zksync_core/src/block_reverter/mod.rs

```diff
@@ -96,13 +100,15 @@ pub struct BlockReverter {
 
 impl BlockReverter {
     pub fn new(
+        main_node: bool,
```

Comment:
Bikeshedding: It could make sense to use an enum instead, like `NodeRole::{Main, External}` for increased readability.

Reply:
done